api/src/main/java/org/pmiops/workbench/cdr/CdrDbConfig.java

```diff
@@ -77,11 +85,22 @@ public void CdrDataSource(CdrVersionDao cdrVersionDao) {
       afterPropertiesSet();
     }
 
+    @PostConstruct
+    public void init() {
+      finishedInitialization = true;
```

Comment:
How many instantiations of `CdrDataSource` will exist? Are they accessed concurrently? Would `init()` and `determineCurrentLookupKey()` be called concurrently?

Reply:
There's only one. This method should be called once after initialization is done, but before anybody actually gets to use it, I believe. I don't know whether init() could be called concurrently with determineCurrentLookupKey()... I don't expect that will happen.

Reply:
OK, if there's only one then I'm less concerned, likely the initialization is happening on a single Spring startup thread.